odoo/models.py

```diff
@@ -3447,9 +3447,9 @@ def _write(self, vals):
                     # synchronize translated terms when possible.
                     self.env['ir.translation']._sync_terms_translations(field, self)
 
-                elif has_translation and field.translate:
+                elif self.env.lang and field.translate:
```

Comment:
Is it intended that when the lang is set to en_US we'll both change the base term (line 3409) and set a translation?

Reply:
Yes, this is more a bugfix IMHO. In case you have an English translation, making a write was only writing on the master record but the translation was not updated. That was revealed by 73a7534bfc0e098a but could potentially be reproduced in stable so could be worth backporting.